src/lib/mdns/Advertiser.h

```diff
@@ -161,13 +191,18 @@ class CommissionAdvertisingParameters : public BaseAdvertisingParams<CommissionA
     CommssionAdvertiseMode GetCommissionAdvertiseMode() const { return mMode; }
 
 private:
-    uint8_t mShortDiscriminator  = 0;
-    uint16_t mLongDiscriminator  = 0; // 12-bit according to spec
-    CommssionAdvertiseMode mMode = CommssionAdvertiseMode::kCommissioning;
+    uint8_t mShortDiscriminator       = 0;
+    uint16_t mLongDiscriminator       = 0; // 12-bit according to spec
+    CommssionAdvertiseMode mMode      = CommssionAdvertiseMode::kCommissionableNode;
+    bool mCommissioningModeEnabled    = false;
+    bool mOpenWindowCommissioningMode = false;
     chip::Optional<uint16_t> mVendorId;
     chip::Optional<uint16_t> mProductId;
+    chip::Optional<uint16_t> mDeviceType;
+    chip::Optional<const char *> mDeviceName;
```

Comment:
Add required lifetime documentation for the " const char *".

Reply:
My thoughts were that this requires lifetime where as the other types do not as they have their value inside the object itself.

I believe you would be the first to add such documentation.

Reply:
In general we've been trying to avoid char *s, and instead put it on the stack. Maybe there's a way to move this to a char[] with some constant size (like below)

Reply:
Done in latest rev